tests/integration/dynamic_ip_test.py

```diff
@@ -576,6 +606,71 @@ def test_ipv6_autoconf_only(dhcpcli_up):
     libnmstate.apply(desired_state)
 
 
+@pytest.mark.xfail(reason='DHCP server does not run behind vlan', strict=True)
+def test_linux_bridge_over_vlan_over_bond_over_slave_single_transaction(
+    bond0, bond0_vlan101
+):
+    bridge_name = TEST_BRIDGE
+    vlan_ifname = bond0_vlan101[Interface.KEY][0][Interface.NAME]
+    bridge_config_state = add_port_to_bridge(
+        create_bridge_subtree_state(), vlan_ifname, BRIDGE_PORT_CONFIG
+    )
+
+    bridge_ip_config = {
+        Interface.IPV4: create_ipv4_state(enabled=True, dhcp=True)
+    }
+    with linux_bridge(
+        bridge_name,
+        bridge_config_state,
+        extra_iface_state=bridge_ip_config,
+        create=False,
+    ) as bridge0:
+        desired_state = bond0
+        vlan_iface_state = bond0_vlan101[Interface.KEY][0]
+        desired_state[Interface.KEY].append(vlan_iface_state)
+        bridge_iface_state = bridge0[Interface.KEY][0]
+        desired_state[Interface.KEY].append(bridge_iface_state)
```

Comment:
Could you extract it to a `state_merge` helper function?
It will also be helpful when reading to name the state variable with what it includes (e.g. bridge_vlan_bond_with_ipv4_state).